src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -263,12 +267,23 @@ private DiagnosticMode GetDiagnosticMode(RequestContext context)
         {
             using var _ = ArrayBuilder<LSP.Diagnostic>.GetInstance(out var result);
             var diagnostics = await diagnosticSource.GetDiagnosticsAsync(DiagnosticAnalyzerService, context, cancellationToken).ConfigureAwait(false);
-            context.TraceInformation($"Found {diagnostics.Length} diagnostics for {diagnosticSource.GetDocumentIdentifier().Uri} in {diagnosticSource.GetProject().Name}");
+
+            // If we can't get a text document identifier we can't report diagnostics for this source.
+            // This can happen for 'fake' projects (e.g. used for TS script blocks).
+            var documentIdentifier = diagnosticSource.GetDocumentIdentifier();
+            if (documentIdentifier == null)
+            {
+                // We are not expecting to get any diagnostics for sources that don't have a path.
+                Contract.ThrowIfFalse(diagnostics.IsEmpty);
+                return default;
```

Comment:
You should be able to write 'null' here -- otherwise this felt a bit odd that TReport might be a struct or something and your null check might do something off.

Reply:
I tried this - even constraining TReport to class.  But it always gave me an error.  Can try again though.

Reply:
I take it back.  There is an implementation that does use struct (for SumType).  I have refactorted this to avoid the issue entirely.